frontend/packages/console-app/src/components/nav/NavHeader.tsx

```diff
@@ -111,7 +111,12 @@ const NavHeader: React.FC<NavHeaderProps> = ({ onPerspectiveSelected }) => {
           </MenuToggle>
         )}
         popperProps={{
-          appendTo: () => document.querySelector("[data-test-id='perspective-switcher-toggle']"),
+          appendTo: () => {
+            const element = document.querySelector<HTMLElement>(
+              "[data-test-id='perspective-switcher-toggle']",
+            );
+            return element || document.body;
+          },
```

Comment:
does e2e still work with this change?

```suggestion
          appendTo: "inline"
```